src/ir/auth_logic/ast.h

```diff
@@ -93,32 +93,25 @@ class BaseFact {
 class CanSay;
 class Fact;
 
-// CanSay corresponds to an expression of the form <principal> canSay Fact
 class CanSay {
  public:
-  explicit CanSay(Principal principal, std::unique_ptr<Fact>& fact)
-      : principal_(principal), fact_(std::move(fact)) {}
+  explicit CanSay(Principal principal) : principal_(std::move(principal)) {}
   const Principal& principal() const { return principal_; }
-  const Fact* fact() const { return fact_.get(); }
 
  private:
   Principal principal_;
-  std::unique_ptr<Fact> fact_;
 };
```

Comment:
`CanSay` is looking pretty thin. Would it make more sense for `Fact` to just contain a list of `Principal`s with the list name indicating that they are `CanSay`?

Reply:
I feel CanSay will be easy to understand that Fact is a combination of CanSay facts and Base Facts. Removing canSay and just keeping Principal would need someone new to look at grammar to understand Fact.

Reply:
+1 to just having a list of `Principal`s in `Fact` and getting rid of the `CanSay` class.

Reply:
Done!